elastic/eui
# [Emotion] Convert EuiContextMenu, EuiContextMenuPanel, and EuiContextMenuItem

## Summary

closes https://github.com/elastic/eui/issues/6392

Converting this component to CSS-in-JS also will also address https://github.com/elastic/kibana/issues/169571 and https://github.com/elastic/kibana/pull/169313#issuecomment-1776244166 (once merged into Kibana main).

This is kind of a lot to review (especially tests, which I went in and edited judiciously), so I think I'd recommend sticking to QA only for reviewing this PR (as regression/QA testing is more important here)

Also while here, I removed an extra unnecessary `<div>` wrapper within `EuiContextMenuItem`. Div wrappers left me an orphan in a dark alley at a young age, and this is how I seek justice.

## QA

- Go to https://eui.elastic.co/pr_7312/#/navigation/context-menu
- [x] Regression test that all context menus look the same as production
    - ⚠️ NOTE: There is one (intentional) change, which is that context menu `title`s are now slightly less tall than before. This is expected, and due to changes in `line-height` from new `euiTitle` Emotion styles.
- Go to https://eui.elastic.co/pr_7312/#/layout/bottom-bar
- Click the first toggle example
- Click the `Help` button
- [x] Confirm that the resulting context menu popover renders with the correct text color in both light and dark mode

### General checklist

- Browser QA
- Docs site QA - N/A
- Code quality checklist
- Release checklist
    ~- [ ] If applicable, added the **breaking change** issue label (and filled out the breaking change checklist)~
- Designer checklist - N/A

<details><summary><h3>Emotion checklist</h3></summary>

**Kibana usage**

- [x] Search Kibana's codebase for `{euiComponent}-` (case sensitive) to check for usage of modifier classes - _None_
    - ⚠️ [1 usage](https://github.com/elastic/kibana/blob/2207928e63fa3900c13f9a8b76a0e5c2e59bfa92/x-pack/test/functional/page_objects/observability_log_explorer.ts#L245)
- [x] Any Sass or CSS that will need to be updated, particularly if a component Sass var was deleted - _None_
     - ⚠️ Yes, [1 usage](https://github.com/elastic/kibana/blob/2ddc342884a6f3b6302b077f89aecf59c3d9dbd6/x-pack/plugins/maps/public/connected_components/mb_map/tooltip_control/features_tooltip/features_tooltip.tsx#L123-L137)

---

**General**

- ~[ ] Checked component playground~ No playground

---

**Unit tests**

- [x] Enzyme tests converted to RTL

---

**Sass/Emotion conversion process**

- ~[ ] All SCSS overrides have been removed from the Amsterdam theme~ _No overrides_

---

**CSS tech debt**


---

**DOM Cleanup**

- [x] Did **NOT** remove any block/element classNames (e.g. `euiComponent`, `euiComponent__child`) - _Note: removed an entire child node/div wrapper_
- [x] **SEARCH KIBANA FIRST**: Deleted any modifier classNames or maps if not being used in Kibana. - _No usages_

---

**Extras/nice-to-have**

</details>
